packages/nc-gui/components/ProjectTreeView.vue

```diff
@@ -963,7 +985,90 @@ export default {
     openLink(link) {
       window.open(link, '_blank');
     },
-
+    async checkAndDeleteTable(table) {
+      const meta = this.$store.state.meta.metas[table.tn] || await this.loadTableSchema(table);
+      if (
+        !meta || (
+          (meta.hasMany && meta.hasMany.length) ||
+          (meta.manyToMany && meta.manyToMany.length) ||
+          (meta.belongsTo && meta.belongsTo.length))
+      ) {
+        return this.$toast.info('Please delete relations before deleting table.').goAway(3000)
+      }
+      await this.deleteTable('showDialog', table._nodes)
+    },
+    async loadTableSchema(table) {
+      return await this.$store.dispatch('meta/ActLoadMeta', {
+        env: table._nodes.env,
+        dbAlias: table._nodes.dbAlias,
+        tn: table.tn
+      })
+    },
+    async deleteTable(action = '', nodes=null) {
+      if(nodes) this.dialogDeleteTable.nodes = nodes;
+      if (action === 'showDialog') {
+        this.dialogDeleteTable.dialogShow = true
+      } else if (action === 'hideDialog') {
+        this.dialogDeleteTable.dialogShow = false
+      } else {
+        nodes = this.dialogDeleteTable.nodes;
+        let relationListAll = await this.$store.dispatch('sqlMgr/ActSqlOp', [{
+          env: nodes.env,
+          dbAlias: nodes.dbAlias
+        }, 'relationListAll'])
+        relationListAll = relationListAll.data.list.filter(rel => rel.rtn === nodes.tn).map(({ tn }) => tn)
+        if (relationListAll.length) {
+          this.$toast.info('Table can\'t be  deleted  since Table is being referred in following tables : ' + relationListAll.join(', ')).goAway(10000)
```

Comment:
```suggestion
          this.$toast.info(`Table can\'t be  deleted  since Table is being referred in following tables :   ${relationListAll.join(', ')}`).goAway(10000)
```